### DISCUSSION TITLE
Google Play rejecting the app while giving error on ad consent SDK.

### DISCUSSION THREAD

**Discussion by wasiquehaider** [invertase/react-native-firebase#5384] [Q&A]
I have tried everything, tried latest version of firebase still google keeps rejecting the apk. Did anyone face thiesame issue  ?  I am attaching the error email for you to take a deeper look into this issue.

"dependencies": {
    "@react-native-community/async-storage": "^1.10.0",
    "@react-native-community/audio-toolkit": "^2.0.3",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-community/netinfo": "^5.8.1",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "@react-native-community/viewpager": "^3.3.0",
    "@react-native-firebase/admob": "^11.4.1",
    "@react-native-firebase/app": "^11.4.1",
    "@react-native-firebase/crashlytics": "^11.4.1",
    "@react-native-firebase/messaging": "^11.4.1",
    "@react-navigation/native": "^5.1.6",
    "@react-navigation/stack": "^5.2.13",
    "@sentry/react-native": "^2.2.1",
    "apisauce": "^1.1.1",
    "async": "^3.2.0",
    "lodash": "^4.17.15",
    "moment": "^2.24.0",
    "react": "16.13.1",
    "react-native": "0.63.3",
    "react-native-animatable": "^1.3.3",
    "react-native-bootsplash": "^2.2.4",
    "react-native-device-info": "^7.0.2",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.6.1",
    "react-native-iap": "^5.2.14",
    "react-native-inappbrowser-reborn": "^3.5.1",
    "react-native-indicators": "^0.17.0",
    "react-native-keyboard-manager": "^4.0.13-16",
    "react-native-modal": "^11.5.6",
    "react-native-offline": "^5.6.1",
    "react-native-orientation-locker": "^1.1.8",
    "react-native-popup-dialog": "^0.18.3",
    "react-native-purchases": "^3.2.0",
    "react-native-push-notification": "^7.0.0",
    "react-native-reanimated": "^1.8.0",
    "react-native-safe-area-context": "^0.7.3",
    "react-native-screens": "^2.5.0",
    "react-native-share": "^3.3.2",
    "react-native-snap-carousel": "^3.9.0",
    "react-native-system-setting": "^1.7.6",
    "react-native-webview": "^10.10.0",
    "react-redux": "^7.2.0",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "redux-persist": "^6.0.0",
    "redux-saga": "^1.1.3"
  },

<img width="748" alt="Screenshot 2021-06-02 at 1 08 48 PM" src="https://user-images.githubusercontent.com/26715284/120447409-015f2e80-c3a4-11eb-98be-8ee15439bafd.png">

**Comment by mikehardy** [comment]
You have included none of the information we require to make any sort of informed response.
If you open an issue vs beginning a discussion, and you fill in the template we may be able to help.
The error message does sound pretty informative though, I imagine by examining the dependencies brought in by react-native-firebase modules used on the android side in combination with the google play store information, and using current versions of all the software involved, you may make progress.

**Reply by wasiquehaider** [reply]
It is saying the ad consent sdk is deprecated in the firebase sdk. Is it not updated yet according to the european countries ? Also I have added the dependencies that I am using inside my app.

**Reply by mikehardy** [reply]
Your dependencies are out of date, spending any time troubleshooting out of date dependencies is typically not time well spent.
If they are out of date because you are staying on v11.x here so that you have admob, then I think you have your answer: admob + kids apps == not allowed as far as I know

**Reply by mikehardy** [reply]
Looks like you posted again - I just searched after your last post (that also lacked details so I could not help) and found this. If you want us to help you really must provide us with more information

**Reply by wasiquehaider** [reply]
> Your dependencies are out of date, spending any time troubleshooting out of date dependencies is typically not time well spent. If they are out of date because you are staying on v11.x here so that you have admob, then I think you have your answer: admob + kids apps == not allowed as far as I know

our many apps containing kids content are showing ads and they are built on native platforms. They seem to work fine :/

Anyways here is my package.json


> {
  "name": "abcApp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "npx react-native run-android",
    "apk:test:release": "cd android; ./gradlew clean; ./gradlew assembleRelease; cd ..;",
    "apk:test:debug": "npx react-native run-android --variant=release",
    "ios": "npx react-native run-ios",
    "start": "npx react-native start",
    "reconnect": "adb reverse tcp:8081 tcp:8081",
    "gradlewClean": "cd android; ./gradlew clean; cd ..;",
    "podInstall": "cd ios; pod install; cd ..;",
    "resetPackager": "npx react-native start --reset-cache",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-native-community/async-storage": "^1.10.0",
    "@react-native-community/audio-toolkit": "^2.0.3",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-community/netinfo": "^5.8.1",
    "@react-native-community/push-notification-ios": "^1.8.0",
    "@react-native-community/viewpager": "^3.3.0",
    "@react-native-firebase/app": "^11.4.1",
    "@react-native-firebase/crashlytics": "^11.4.1",
    "@react-native-firebase/messaging": "^11.4.1",
    "@react-navigation/native": "^5.1.6",
    "@react-navigation/stack": "^5.2.13",
    "@sentry/react-native": "^2.2.1",
    "apisauce": "^1.1.1",
    "async": "^3.2.0",
    "lodash": "^4.17.15",
    "moment": "^2.24.0",
    "react": "16.13.1",
    "react-native": "0.63.3",
    "react-native-admob": "^2.0.0-beta.6",
    "react-native-animatable": "^1.3.3",
    "react-native-bootsplash": "^2.2.4",
    "react-native-device-info": "^7.0.2",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.6.1",
    "react-native-iap": "^5.2.14",
    "react-native-inappbrowser-reborn": "^3.5.1",
    "react-native-indicators": "^0.17.0",
    "react-native-keyboard-manager": "^4.0.13-16",
    "react-native-modal": "^11.5.6",
    "react-native-offline": "^5.6.1",
    "react-native-orientation-locker": "^1.1.8",
    "react-native-popup-dialog": "^0.18.3",
    "react-native-purchases": "^3.2.0",
    "react-native-push-notification": "^7.0.0",
    "react-native-reanimated": "^1.8.0",
    "react-native-safe-area-context": "^0.7.3",
    "react-native-screens": "^2.5.0",
    "react-native-share": "^3.3.2",
    "react-native-snap-carousel": "^3.9.0",
    "react-native-system-setting": "^1.7.6",
    "react-native-webview": "^10.10.0",
    "react-redux": "^7.2.0",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "redux-persist": "^6.0.0",
    "redux-saga": "^1.1.3"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/runtime": "^7.8.4",
    "@react-native-community/eslint-config": "^1.1.0",
    "babel-jest": "^25.1.0",
    "eslint": "^6.5.1",
    "jest": "^25.1.0",
    "metro-react-native-babel-preset": "^0.59.0",
    "react-test-renderer": "16.13.1"
  },
  "jest": {
    "preset": "react-native"
  }
}

**Reply by mikehardy** [reply]
So, why not use updated versions here? 12.8.0 is current?

**Reply by wasiquehaider** [reply]
I have updated firebase to the latest version 12.8.0 but it was incompatible with the firebase/admob package thats why I needed to remove that and added react native admob and deployed the app but they rejected it again.

**Reply by mikehardy** [reply]
I don't believe react-native-admob is updated either, though I may be wrong? This may just be a case of "all the software I get for free hasn't been updated yet" - not saying that to be harsh, just saying it takes time to get things all up to date and most of these things are done in spare time between paying work (in react-native-admob's case). In our case here we are working on it and we do provide official firebase support but firebase removed admob, so it gets a little less attention unfortunately.

We are working on it though #5329 - I'm hoping to have an up to date admob package "soon" but it's not easy and it is in time between other projects

**Reply by wasiquehaider** [reply]
so how do I clear the playstore rejection in the mean time ?

**Reply by mikehardy** [reply]
I'm not sure it's possible, either remove ads I guess or be the person that contributes their time to forward porting react-native-admob

**Comment by nguyenthixanh18011988** [comment]
me too. have you update the app?

**Reply by wasiquehaider** [reply]
Not really it is still rejecting the app.
